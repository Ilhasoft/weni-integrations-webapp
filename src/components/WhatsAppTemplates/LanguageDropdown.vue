<template>
  <unnnic-dropdown>
    <div slot="trigger" class="template-language-dropdown">
      <unnnic-icon-svg
        class="template-language-dropdown__icon"
        :icon="getTranslationStatusIcon(template.translations[0])"
        size="sm"
        :scheme="getTranslationStatusColor(template.translations[0])"
      />
      {{ getTemplateDefaultLanguage(template) }}
      <unnnic-icon-svg
        class="template-language-dropdown__icon"
        icon="arrow-button-down-1"
        size="sm"
      />
    </div>
    <unnnic-dropdown-item
      class="template-language-dropdown__item"
      v-for="(translation, index) in template.translations"
      :key="index"
    >
      <unnnic-icon-svg
        class="template-language-dropdown__item__icon"
        :icon="getTranslationStatusIcon(translation)"
        size="sm"
        :scheme="getTranslationStatusColor(translation)"
      />
      {{ translation.language }}
    </unnnic-dropdown-item>
  </unnnic-dropdown>
</template>

<script>
  export default {
    name: 'LanguageDropdown',
    props: {
      template: {
        type: Object,
        required: true,
      },
    },
    data() {
      return {
        translationStatusMap: {
          APPROVED: {
            icon: 'check-circle-1-1-1',
            color: 'feedback-green',
          },
        },
      };
    },
    methods: {
      getTemplateDefaultLanguage(template) {
        return template.translations[0].language;
      },
      getTranslationStatusIcon(translation) {
        return this.translationStatusMap[translation.status]?.icon || 'delete-1-1';
      },
      getTranslationStatusColor(translation) {
        return this.translationStatusMap[translation.status]?.color || 'feedback-red';
      },
    },
  };
</script>

<style lang="scss" scoped>
  .template-language-dropdown {
    display: flex;
    flex-wrap: nowrap;
    justify-content: space-between;
    border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
    border-radius: $unnnic-border-radius-pill;
    padding: $unnnic-spacing-stack-nano $unnnic-spacing-inline-xs;
    align-items: center;
    width: 100px;
    text-transform: uppercase;
    cursor: pointer;

    &__icon {
      margin-right: $unnnic-spacing-inline-nano;
    }

    &__item {
      width: 80px;
      text-transform: uppercase;

      &__icon {
        margin-right: $unnnic-spacing-inline-nano;
      }
    }
  }
</style>
