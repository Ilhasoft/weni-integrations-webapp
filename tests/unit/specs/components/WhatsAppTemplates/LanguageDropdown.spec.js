import { shallowMount, createLocalVue } from '@vue/test-utils';
import LanguageDropdown from '@/components/WhatsAppTemplates/LanguageDropdown.vue';

const localVue = createLocalVue();

describe('components/WhatsAppTemplates/LanguageDropdown.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(LanguageDropdown, {
      localVue,
      stubs: {
        UnnnicDropdown: true,
        UnnnicIconSvg: true,
        UnnnicDropdownItem: true,
      },
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        template: {
          translations: [
            { language: 'en', status: 'APPROVED' },
            { language: 'es', status: 'REJECTED' },
          ],
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
