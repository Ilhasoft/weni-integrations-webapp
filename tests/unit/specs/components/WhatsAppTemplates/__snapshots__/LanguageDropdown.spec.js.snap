// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WhatsAppTemplates/LanguageDropdown.vue should be rendered properly 1`] = `
<unnnicdropdown-stub>
  <div class="template-language-dropdown">
    <unnniciconsvg-stub icon="check-circle-1-1-1" size="sm" scheme="feedback-green" class="template-language-dropdown__icon"></unnniciconsvg-stub>
    en
    <unnniciconsvg-stub icon="arrow-button-down-1" size="sm" class="template-language-dropdown__icon"></unnniciconsvg-stub>
  </div>
  <unnnicdropdownitem-stub class="template-language-dropdown__item">
    <unnniciconsvg-stub icon="check-circle-1-1-1" size="sm" scheme="feedback-green" class="template-language-dropdown__item__icon"></unnniciconsvg-stub>
    en
  </unnnicdropdownitem-stub>
  <unnnicdropdownitem-stub class="template-language-dropdown__item">
    <unnniciconsvg-stub icon="delete-1-1" size="sm" scheme="feedback-red" class="template-language-dropdown__item__icon"></unnniciconsvg-stub>
    es
  </unnnicdropdownitem-stub>
</unnnicdropdown-stub>
`;
